src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts

```diff
@@ -1,33 +1,490 @@
+    showRecipientNameTo: [],
+    isEditable: true,
+    isSaving: false,
+    isCollapsed: false,
+    isVisibilityChanged: false,
+    isFeedbackPathChanged: false,
+    isQuestionDetailsChanged: false,
+  };
+
+  const testQuestionEditFormModel2: QuestionEditFormModel = {
+    feedbackQuestionId: 'feedback-question-2',
+    questionNumber: 2,
+    questionBrief: 'question brief',
+    questionDescription: 'description',
+    isQuestionHasResponses: false,
+    questionType: FeedbackQuestionType.TEXT,
+    questionDetails: {
+      questionType: FeedbackQuestionType.TEXT,
+      questionText: 'question text',
+    } as FeedbackTextQuestionDetails,
+    giverType: FeedbackParticipantType.STUDENTS,
+    recipientType: FeedbackParticipantType.INSTRUCTORS,
+    numberOfEntitiesToGiveFeedbackToSetting: NumberOfEntitiesToGiveFeedbackToSetting.UNLIMITED,
+    customNumberOfEntitiesToGiveFeedbackTo: 5,
+    showResponsesTo: [FeedbackVisibilityType.GIVER_TEAM_MEMBERS, FeedbackVisibilityType.INSTRUCTORS],
+    showGiverNameTo: [],
+    showRecipientNameTo: [],
+    isEditable: true,
+    isSaving: false,
+    isCollapsed: false,
+    isVisibilityChanged: false,
+    isFeedbackPathChanged: false,
+    isQuestionDetailsChanged: false,
+  };
+
+  const sessionEditFormModel: SessionEditFormModel = {
+    courseId: 'testId',
+    timeZone: 'Asia/Singapore',
+    courseName: 'Test Course',
+    feedbackSessionName: 'test session',
+    instructions: 'Instructions',
+
+    submissionStartTime: { hour: 23, minute: 59 },
+    submissionStartDate: { year: 0, month: 0, day: 0 },
+    submissionEndTime: { hour: 23, minute: 59 },
+    submissionEndDate: { year: 0, month: 0, day: 0 },
+    gracePeriod: 0,
+
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    customSessionVisibleTime: { hour: 23, minute: 59 },
+    customSessionVisibleDate: { year: 0, month: 0, day: 0 },
+
+    responseVisibleSetting: ResponseVisibleSetting.CUSTOM,
+    customResponseVisibleTime: { hour: 23, minute: 59 },
+    customResponseVisibleDate: { year: 0, month: 0, day: 0 },
+
+    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+    publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
+
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+
+    templateSessionName: '',
+
+    isSaving: false,
+    isEditable: false,
+    isDeleting: false,
+    isCopying: false,
+    hasVisibleSettingsPanelExpanded: false,
+    hasEmailSettingsPanelExpanded: false,
+  };
+
   let component: InstructorSessionEditPageComponent;
   let fixture: ComponentFixture<InstructorSessionEditPageComponent>;
+  let feedbackSessionsService: FeedbackSessionsService;
+  let feedbackQuestionsService: FeedbackQuestionsService;
+  let courseService: CourseService;
+  let studentService: StudentService;
+  let instructorService: InstructorService;
+  let navigationService: NavigationService;
+  let ngbModal: NgbModal;
+  const mockModalRef: MockNgbModalRef = new MockNgbModalRef();
```

Comment:
We can shift this to within the test case itself, seems it looks like it's used only there